api/client/commands.go

```diff
@@ -162,21 +163,19 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 			}
 
 			// Verify that 'filename' is within the build context
-			if !strings.HasSuffix(absRoot, string(os.PathSeparator)) {
-				absRoot += string(os.PathSeparator)
-			}
-			if !strings.HasPrefix(filename, absRoot) {
-				return fmt.Errorf("The Dockerfile (%s) must be within the build context (%s)", *dockerfileName, root)
+			filename, err = symlink.FollowSymlinkInScope(filename, absRoot)
```

Comment:
Just to be sure: this *does* break an existing use case (where the Dockerfile would be a symlink to something outside of scope)? I haven't followed all IRC discussions but I assume this was agreed on.